openedx/core/djangoapps/external_user_ids/tests/test_batch_generate_id.py

```diff
@@ -0,0 +1,67 @@
+"""
+Test batch_get_or_create in ExternalId model
+"""
+
+from django.test import TransactionTestCase
+from common.djangoapps.student.tests.factories import UserFactory
+from openedx.core.djangoapps.external_user_ids.models import ExternalId
+from openedx.core.djangoapps.external_user_ids.tests.factories import ExternalIDTypeFactory
+
+
+class TestBatchGenerateExternalIds(TransactionTestCase):
+    """
+    Test ExternalId.batch_get_or_create_user_ids
+    """
+
+    # Following are the queries
+    # 1 - Get ExternalIdType
+    # 2 - Find users for those external ids needs to be created
+    # 3 - Get external ids that already exists
+    # 4 - BEGIN (from bulk_create)
+    # 5 - Create new external ids
+    EXPECTED_NUM_OF_QUERIES = 5
+
+    def test_batch_get_or_create_user_ids(self):
+        """
+        Test if batch_get_or_create creates ExternalIds in batch
+        """
+        id_type = ExternalIDTypeFactory.create(name='test')
+        users = [UserFactory() for _ in range(10)]
+
+        with self.assertNumQueries(self.EXPECTED_NUM_OF_QUERIES):
+            result = ExternalId.batch_get_or_create_user_ids(users, id_type)
+
+        assert len(result) == len(result)
```

Comment:
This surely is not what you meant!
```suggestion
        assert len(result) == len(users)
```

Reply:
Sorry for that and thanks for catching it.